bluemira/equilibria/flux_surfaces.py

```diff
@@ -872,3 +876,251 @@ def poloidal_angle(Bp_strike, Bt_strike, gamma):
+    Construct matrix from harmonic amplitudes at given points (in spherical coords).
+
+    The matrix is used in a spherical harmonic approximation of the vacuum/coil
+    contribution to the poilodal flux (psi):
+
+        psi = SUM(
+            A_l * ( r**(l+1) / r_t**l ) * sin (theta) *
+            ( P_l * cos(theta_f) / sqrt(l*(l+1)) )
+        )
+
+    Where l = degree, A_l are the spherical harmonic coeffcients/ampletudes,
+    and is P_l * cos(theta_f) are the associated Legendre polynomials of
+    degree l and order (m) = 1.
+
+    N.B. Vacuum Psi = Total Psi - Plasma Psi.
+
+    Parameters
+    ----------
+    collocation_r: np.array
+        R values of collocation points
+    collocation_theta: np.array
+        Theta values of collocation points
+    n_collocation: integer
+        Number of collocation points
+    max_degree: integer
+        Maximum degree of harmonic to calculate up to
+    r_t: float
+        Typical length scale (e.g. radius at outer midplane)
+
+    Returns
+    -------
+    harmonics2collocation: np.array
+        Matrix of harmonic amplitudes (to get spherical harmonic coefficents
+        use matrix @ coefficents = vector psi_vacuum at colocation points)
+    """
+    # [number of points, number of degrees]
+    harmonics2collocation = np.zeros([n_collocation, max_degree])
+    # First 'harmonic' is constant (this line avoids Nan isuues)
+    harmonics2collocation[:, 0] = 1
+
+    # SH coeffcient matrix
+    # SH coeffs = harmonics2collocation \ vector psi_vacuum at colocation points
+    for degree in np.arange(1, max_degree):
+        harmonics2collocation[:, degree] = (
+            collocation_r ** (degree + 1)
+            * np.sin(collocation_theta)
+            * lpmv(1, degree, np.cos(collocation_theta))
+            / ((r_t**degree) * np.sqrt(degree * (degree + 1)))
+        )
+
+    return harmonics2collocation
+
+
+def collocation_points(n_points, x_bdry, z_bdry, point_type):
+    """
+    Create a set of collocation points for use wih spherical harmonic
+    approximations. Points are found within the user-supplied
+    boundary and should correspond to the LCFS of a chosen equilibrium.
+    Curent functionality is for:
+        - equispaced points on an arc of fixed radius,
+        - extrema,
+        - equispaced points on an arc of fixed radius plus extrema,
+
+    Parameters
+    ----------
+    n_points: integer
+        Number of points/targets (not including extrema - these are added
+        automatically if relevent).
+    x_bdry: np.array
+        X location of plamsa bounday
+    z_bdry: np.array
+        Z locations of plamsa boundary
+    point_type: string
+        Method for creating a set of points: 'arc', 'arc_plus_extrema' or 'random'
+
+    Returns
+    -------
+    collocation_r: np.array
+        R values of collocation points
+    collocation_theta: np.array
+        Theta values of collocation points
+    n_collocation: integer
+        Number of collocation points
```

Comment:
Not the return signature. Might one be able to figure out the number of points from the return vector if necessary? :)

Reply:
That sound much too sensible :) but change made.